musicbot/bot.py

```diff
@@ -4159,120 +4150,47 @@ async def on_voice_state_update(self, member, before, after):
-
-        auto_paused = self.server_specific_data[channel.guild]["auto_paused"]
-
         try:
             player = await self.get_player(channel)
         except exceptions.CommandError:
             return
 
-        def is_active(member):
-            if not member.voice:
-                return False
-
-            if any([member.voice.deaf, member.voice.self_deaf, member.bot]):
-                return False
-
-            return True
-
-        if not member == self.user and is_active(member):  # if the user is not inactive
-            if (
-                player.voice_client.channel != before.channel
-                and player.voice_client.channel == after.channel
-            ):  # if the person joined
-                if auto_paused and player.is_paused:
-                    log.info(
-                        autopause_msg.format(
-                            state="Unpausing",
-                            channel=player.voice_client.channel,
-                            reason="",
-                        ).strip()
-                    )
-
-                    self.server_specific_data[player.voice_client.guild][
-                        "auto_paused"
-                    ] = False
-                    player.resume()
-            elif (
-                player.voice_client.channel == before.channel
-                and player.voice_client.channel != after.channel
-            ):
-                if not any(
-                    is_active(m) for m in player.voice_client.channel.members
-                ):  # channel is empty
-                    if not auto_paused and player.is_playing:
-                        log.info(
-                            autopause_msg.format(
-                                state="Pausing",
-                                channel=player.voice_client.channel,
-                                reason="(empty channel)",
-                            ).strip()
-                        )
+        autopause_msg = "{state} in {channel.guild.name}/{channel.name} {reason}"
+        auto_paused = self.server_specific_data[channel.guild]["auto_paused"]
 
-                        self.server_specific_data[player.voice_client.guild][
-                            "auto_paused"
-                        ] = True
-                        player.pause()
-            elif (
-                player.voice_client.channel == before.channel
-                and player.voice_client.channel == after.channel
-            ):  # if the person undeafen
-                if auto_paused and player.is_paused:
-                    log.info(
-                        autopause_msg.format(
-                            state="Unpausing",
-                            channel=player.voice_client.channel,
-                            reason="(member undeafen)",
-                        ).strip()
-                    )
+        def set_autopause(value):
+            self.server_specific_data[player.voice_client.guild]["auto_paused"] = value
```

Comment:
This can probably be removed, its only being called once so defeats the line save arguement for nested functions.

Reply:
Good catch, that was called twice in a previous version, no need to keep it around